### PR TITLE
use http vs curl

### PR DESCRIPTION
**Pull request by jsncmgs1**
Removes curl and use http.Get.

There's probably a world where we drop shelling to tar and using go's lib, but that can come later.

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -98,13 +99,15 @@ func NameHandler(w http.ResponseWriter, r *http.Request) {
 
 	url := fmt.Sprintf("https://buildpack-registry.s3.amazonaws.com/buildpacks/%s.tgz", id)
 	log.Infof("at=download file=%s url=%s", shimmedBuildpack, url)
-	cmd := fmt.Sprintf(`curl --retry 3 --silent --location "%s" | tar xzm -C %s`, url, target_dir)
```
Comment (jsncmgs1) [inline server/main.go:101]: retry is and redirect follow is built in to go's client, so we get these flags for free.

File: server/main.go [hunk 2]
```diff
@@ -98,13 +99,15 @@ func NameHandler(w http.ResponseWriter, r *http.Request) {
...
-
-	_, err = exec.Command("bash", "-c", cmd).Output()
+	bp, err = downloadBuildpack(url)
+	handlePanic(err)
+	tar := fmt.Sprintf(`tar xzf %s -C %s`, bp, target_dir)
+	_, err = exec.Command("bash", "-c", tar).Output()
 	handlePanic(err)
+	handlePanic(os.Remove(bp))
```
Comment (jsncmgs1) [inline server/main.go:107]: our tar is also including the tgz file into the target dir, I don't know why it didn't do that in the `curl ... | tar -C` command but this was the quickest and easiest way to get it out. 

If someone's tar foo is better than mine I'd like to remove this but it was becoming a time sink.

File: server/main.go [hunk 3]
```diff
@@ -129,3 +132,24 @@ func handlePanic(e error) {
...
 }
+
+func downloadBuildpack(url string) (string, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	id := uuid.New().String() + ".tgz"
```
Comment (jabrown85) [inline server/main.go:143]: Maybe something like `ioutil.TempFile("", "*.tgz")` would read better

File: server/main.go (hunk 2 above)
Comment (jabrown85) [inline server/main.go:107]: I'd like to see `defer os.Remove(bp)` after `downloadBuildpack` to ensure we attempt to remove, even if the tar fails.

File: go.sum [hunk 4]
```diff
@@ -43,6 +43,8 @@ github.com/onsi/gomega v1.4.3 h1:RE1xgDvH7imwFD45h+u2SgIfERHlS2yNG4DObb5BSKU=
 github.com/onsi/gomega v1.4.3/go.mod h1:ex+gbHU/CVuBBDIJjb2X0qEXbFg53c61hWP/1CpauHY=
 github.com/onsi/gomega v1.5.0 h1:izbySO9zDPmjJ8rDjLvkA2zJHIo+HkYXHnf7eN7SSyo=
 github.com/onsi/gomega v1.5.0/go.mod h1:ex+gbHU/CVuBBDIJjb2X0qEXbFg53c61hWP/1CpauHY=
+github.com/pkg/errors v0.8.1 h1:iURUrRGxPUNPdy5/HRSm+Yj6okJ6UtLINN0Q9M4+h3I=
```
Comment (jabrown85) [inline go.sum:46]: `go mod tidy`?

**Review by jabrown85 — COMMENTED** [review]
This looks nice. Couple of nits. Did you compare a buildpack produced by this to see if the contents of a buildpack produced are the same as prod today? I want to be sure we are producing the expected result.

File: server/main.go (hunk 2 above)
Comment (jsncmgs1) [reply server/main.go:107]: @jabrown85 this whole dir we're working in gets nuked here: https://github.com/heroku/cnb-shim/blob/master/server/main.go#L66

**Comment by jsncmgs1** [comment]
> This looks nice. Couple of nits. Did you compare a buildpack produced by this to see if the contents of a buildpack produced are the same as prod today? I want to be sure we are producing the expected result.

I did compare, they're the same.

File: server/main.go (hunk 3 above)
Comment (jsncmgs1) [reply server/main.go]: a1bd62e

File: go.sum (hunk 4 above)
Comment (jsncmgs1) [reply go.sum]: a1bd62e

File: server/main.go (hunk 2 above)
Comment (jabrown85) [reply server/main.go:107]: Then why do `os.Remove(bp)` at all?

**Review by jabrown85 — APPROVED** [review]

**Pull request merged**